core/engine/src/value/inner/nan_boxed.rs

```diff
@@ -19,21 +19,14 @@
 //! | `False`           | `7FF6:0000:0000:0000`    | |
 //! | `True`            | `7FF6:0000:0000:0001`    | |
 //! | `Integer32`       | `7FF7:0000:IIII:IIII`    | 32-bits integer. |
-//! | `BigInt`          | `7FF[8-F]:PPPP:PPPP:PPPP \| 0` | 51-bits pointer. Assumes non-null pointer. |
-//! | `Object`          | `7FF[8-F]:PPPP:PPPP:PPPP \| 1` | 51-bits pointer. |
-//! | `Symbol`          | `7FF[8-F]:PPPP:PPPP:PPPP \| 2` | 51-bits pointer. |
-//! | `String`          | `7FF[8-F]:PPPP:PPPP:PPPP \| 3` | 51-bits pointer. |
+//! | `BigInt`          | `7FF8:PPPP:PPPP:PPPP`    | 49-bits pointer. Assumes non-null pointer. |
+//! | `Object`          | `7FFA:PPPP:PPPP:PPPP`    | 49-bits pointer. |
+//! | `Symbol`          | `7FFC:PPPP:PPPP:PPPP`    | 49-bits pointer. |
+//! | `String`          | `7FFE:PPPP:PPPP:PPPP`    | 49-bits pointer. |
```

Comment:
So in the pointer space we only have 1 unused slot left right which is `7FF8`

Reply:
`7FF8` (or `b0111_1111_1111_1000`) is used by bigint if the pointer is non-null. The 16th MSB isn't used (yet?), so pointers are actually 48 bits (I'll correct the comment).

Reply:
Let me know if the commit helps clarifying this.

Reply:
Yes sorry I meant to ask if 7FFF is unused and reserved